std/experimental/allocator/package.d

```diff
@@ -560,6 +560,178 @@ unittest
     test(theAllocator);
 }
 
+// the class repository used in factory()
+__gshared package TypeInfo_Class[string] _factoryClasses;
```

Comment:
There's no reason to use `__gshared`, instead of `shared` here. Further more if you had used `shared` the compiler would have disallowed this code because it has a race-condition. I will propose an alternative that allows safe concurrent access shortly.